File: 2023/trebuchet/part1.js
const fs = require("fs");

const lines = fs.readFileSync("input.txt", "utf8").split(/\n/);

let sum = 0;
for (const line of lines) {
  const numbers = line.match(/\d/g);

  if (!numbers) {
    continue;
  }

  let firstNumber = numbers[0];
  let lastNumber = numbers[numbers.length - 1];

  sum += +firstNumber * 10 + +lastNumber;
}

console.log(sum);

File: 2023/trebuchet/part1.py
lines: list[str] = []
with open("input.txt") as file:
    lines = file.readlines()

digits: list[int] = []
for line in lines:
    numbers_in_line = [letter for letter in line if letter.isdigit()]

    if numbers_in_line:
        first_number = numbers_in_line[0]
        last_number = numbers_in_line[-1]
        digits.append(int(first_number + last_number))

print(sum(digits))
